### ISSUE TITLE
Option for versioning and updating config files

### ISSUE THREAD

**Issue by Licenser** [basho/cuttlefish#176]
I am not sure if this is a concern for anyone else so I wanted to discuss this before storming off and and creating a pull request ;)

If a config file changes between version 1 and 2 it is quite a pain for the end user to update that file, it means manually transferring every value they changes in the new file.

Usually updates do not change much a option gets added or removed, sometimes something gets renamed.

The simplest idea would be:

1) have cuttlefish reads the old data ignoring missing options and tossing away unknown ones
2) write a new config based on the supplied values or defaults from the new schema files

This would probably cover 90% of all cases, it would fail for cases where the type caged but it would give a good starting approach I think.

This could possibly be handled by a simple bash script? Not sure.

A more complex approach would be versioning the config files, allowing to provide versions of the schema files and additional translations form V1 to V2 to V3 etc. This is a lot more complex but would allow for covering every possible aspect.

**Comment by seancribbs** [comment]
@Licenser Yes, we need to have a plan to deprecate, rename, and remove settings gracefully. Any ideas are appreciated, let's open the discussion.

**Comment by Licenser** [comment]
This is a rather dirty hack bit it's a start. Basically the way I handle the variables right now is that the 'package' that installs only provides a conf.example and not the actual conf file so I'd call this like:

./update_conf.sh sniffle.conf sniffle.conf.example > sniffle.conf.new && mv sniffle.conf sniffle.conf.old && mv sniffle.conf.new sniffle.conf

``` bash
#!/usr/bin/env bash

example=$1
old=$2
get() {
        var=$2
        conf=$1

        res=$(grep "^\\s*${var}\\s*=" $conf | sed "s/^\\s*${var}\\s*=\\s*//")
        if [ -z "${res}" ]
        then
                exit 1
        fi
        echo "${res}"
}


while read line
do
        if echo "${line}" | grep -v '^#' | grep -v '^$' > /dev/null
        then
                key=$(echo "${line}" | sed 's/\s*=.*//')
                if val=$(get ${old} ${key})
                then
                        echo "${key} = ${val}"

                else
                        echo "${line}"
                fi
        else
                echo "${line}"

        fi
done < "${example}"
```

(edited to remove debug prints in the bash script)
